zipkin-lens/src/containers/TracePage/TracePageContainer.js

```diff
@@ -13,31 +13,31 @@
  */
 import { connect } from 'react-redux';
 
-import { fetchTrace } from '../../actions/trace-action';
+import { loadTrace } from '../../actions/trace-action';
 import TracePage from '../../components/TracePage';
-import { treeCorrectedForClockSkew, detailedTraceSummary } from '../../zipkin';
 
-const mapStateToProps = (state, ownProps) => {
-  const props = {
-    isLoading: state.trace.isLoading,
-    traceId: ownProps.match.params.traceId,
-  };
-  if (state.trace.trace.length === 0) {
-    props.traceSummary = null;
-  } else {
-    const corrected = treeCorrectedForClockSkew(state.trace.trace);
-    props.traceSummary = detailedTraceSummary(corrected);
-  }
-  return props;
-};
+const mapStateToProps = (state, ownProps) => ({
+  traceId: ownProps.match.params.traceId,
+  isLoading: state.trace.isLoading,
+  traceSummary: state.trace.traceSummary,
+  correctedTraceMap: state.traces.correctedTraceMap,
+});
 
 const mapDispatchToProps = dispatch => ({
-  fetchTrace: traceId => dispatch(fetchTrace(traceId)),
+  loadTrace: (traceId, correctedTraceMap) => dispatch(loadTrace(traceId, correctedTraceMap)),
 });
 
 const TracePageContainer = connect(
   mapStateToProps,
   mapDispatchToProps,
+  (stateProps, dispatchProps, ownProps) => ({
```

Comment:
This is **mergeProps**.

https://github.com/reduxjs/react-redux/blob/master/docs/api/connect.md#mergeprops-stateprops-dispatchprops-ownprops--object

Reply:
this could make a good code comment